app/src/main/java/cl/maleb/testcarsales/utils/DateUtils.kt

```diff
@@ -0,0 +1,86 @@
+package cl.maleb.testcarsales.utils
+
+import android.app.DatePickerDialog
+import android.content.Context
+import android.os.Build
+import java.text.SimpleDateFormat
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+// Calendar variables
+val c = Calendar.getInstance()
+val year = c.get(Calendar.YEAR)
+val month = c.get(Calendar.MONTH)
+val day = c.get(Calendar.DAY_OF_MONTH)
+
+fun initDate(): String {
+    // method that helps to do the first API call. Returns the current date minus one day.
+    val currentDay = day - 1
+    val currentMonth = month
+    val currentYear = year
+    var dateString = ""
+
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val current = LocalDateTime.now().minusDays(1)
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+        dateString = current.format(formatter)
+    } else {
+        val date = Date(currentYear, currentMonth, currentDay)
+        val formatter = SimpleDateFormat("yyyy-MM-dd")
+        dateString = formatter.format(date)
+    }
+
+    return dateString
+}
+
+fun getCurrentDay(): String {
+    var dateString = ""
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val current = LocalDateTime.now()
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+        dateString = current.format(formatter)
+    } else {
+        val date = Date()
+        val formatter = SimpleDateFormat("yyyy-MM-dd")
+        dateString = formatter.format(date)
+    }
```

Comment:
If this methods do the same thing, you can create a general method to handle the dates and use it wherever you need

Reply:
One option is convert this method to an extension of Context, in this way you can use that context to call the method `getString` and use a string resource instead of the hardcoded text

Reply:
Yeah, I replicated the method and forgot to erase the one that I don't even use it